android/src/com/frostwire/android/gui/fragments/SearchFragment.java

```diff
@@ -286,6 +291,65 @@ public void run() {
         list.setAdapter(adapter);
     }
 
+    private void onSearchResults(final List<? extends SearchResult> results) {
+        FilteredSearchResults fsr = adapter.filter((List<SearchResult>) results);
+        final List<SearchResult> filteredList = fsr.filtered;
+        fileTypeCounter.add(fsr);
+        // if it's a fresh search, make sure to clear keyword detector
+        if (adapter.getCount() == 0 && adapter.getKeywordFiltersPipeline().size() == 0) {
+            resetKeywordDetector();
+        }
+
+        updateKeywordDetector(results);
+
+        getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                adapter.addResults(results, filteredList);
+                showSearchView(getView());
+                refreshFileTypeCounters(true);
+            }
+        });
+    }
+
+    private void updateKeywordDetector(final List<? extends SearchResult> results) {
+        Engine.instance().getThreadPool().submit(new Runnable() {
+            @Override
```

Comment:
perhaps this runnable here isn't necessary anymore, as `requestHistogramUpdate()` changed a lot since this was first written